src/SB/Game/zNPCSpawner.cpp

```diff
@@ -177,3 +177,95 @@ U8 zMovePoint::IsOn()
 {
     return this->on;
 }
+
+st_XORDEREDARRAY* zNPCSpawner::FillPending()
+{
+    ClearPending();
+    ReFillPending();
+    return &this->actvlist;
+}
+
+st_XORDEREDARRAY* zNPCSpawner::ReFillPending()
+{
+    s32 var_r28;
+    zNPCCommon* temp_r29;
+    zNPCSpawner* var_r30;
+
+    var_r28 = 0;
+    var_r30 = this;
+    do
+    {
+        temp_r29 = var_r30->npc_owner;
+        if (((zNPCCommon*)var_r30->npc_owner != NULL) && ((s32)temp_r29->flg_vuln == 1))
+        {
+            XOrdAppend(&this->pendlist, (void*)temp_r29);
+            temp_r29->flg_vuln = 2;
+        }
+        var_r28 += 1;
+        var_r30 += 0xC;
+    } while (var_r28 < 0x10);
```

Comment:
This is probably some sort of for loop over an array, with `var_r29` and `var_r30` being compiler generated variables related to indexing an array.